vizro-core/tests/unit/vizro/models/_components/test_aggrid.py

```diff
@@ -0,0 +1,156 @@
+"""Unit tests for vizro.models.AgGrid."""
+
+import dash_ag_grid as dag
+import pytest
+from asserts import assert_component_equal
+from dash import dcc, html
+
+try:
+    from pydantic.v1 import ValidationError
+except ImportError:  # pragma: no cov
+    from pydantic import ValidationError
+
+import vizro.models as vm
+import vizro.plotly.express as px
+from vizro.managers import data_manager
+from vizro.models._action._action import Action
+from vizro.tables import dash_ag_grid
+
+
+@pytest.fixture
+def dash_ag_grid_with_arguments():
+    return dash_ag_grid(data_frame=px.data.gapminder(), defaultColDef={"resizable": False, "sortable": False})
+
+
+@pytest.fixture
+def dash_ag_grid_with_str_dataframe():
+    return dash_ag_grid(data_frame="gapminder")
+
+
+class TestDunderMethodsAgGrid:
+    def test_create_graph_mandatory_only(self, standard_ag_grid):
+        ag_grid = vm.AgGrid(figure=standard_ag_grid)
+
+        assert hasattr(ag_grid, "id")
+        assert ag_grid.type == "ag_grid"
+        assert ag_grid.figure == standard_ag_grid
+        assert ag_grid.actions == []
+
+    @pytest.mark.parametrize("id", ["id_1", "id_2"])
+    def test_create_ag_grid_mandatory_and_optional(self, standard_ag_grid, id):
+        ag_grid = vm.AgGrid(figure=standard_ag_grid, id=id, actions=[])
+
+        assert ag_grid.id == id
+        assert ag_grid.type == "ag_grid"
+        assert ag_grid.figure == standard_ag_grid
+
+    def test_mandatory_figure_missing(self):
+        with pytest.raises(ValidationError, match="field required"):
+            vm.AgGrid()
+
+    def test_failed_ag_grid_with_no_captured_callable(self, standard_go_chart):
+        with pytest.raises(ValidationError, match="must provide a valid CapturedCallable object"):
+            vm.AgGrid(figure=standard_go_chart)
+
+    @pytest.mark.xfail(reason="This test is failing as we are not yet detecting different types of captured callables")
+    def test_failed_ag_grid_with_wrong_captured_callable(self, standard_px_chart):
+        with pytest.raises(ValidationError, match="must provide a valid ag_grid function vm.AgGrid"):
+            vm.AgGrid(figure=standard_px_chart)
+
+    def test_getitem_known_args(self, dash_ag_grid_with_arguments):
```

Comment:
It's only this and the next test in this class that are testing the dunder methods as far as I can tell; the rest are instantiation tests.